### PR TITLE
Store step times in SaveAllSteppingAction

### PR DESCRIPTION
**Pull request by gonzaponte**
Step time is defined as the average between the pre- and post-step time.

### CODE REVIEW

File: source/actions/SaveAllSteppingAction.cc [hunk 1]
```diff
@@ -63,6 +63,8 @@ void SaveAllSteppingAction::UserSteppingAction(const G4Step* step)
 
   G4ThreeVector initial_pos = pre ->GetPosition();
   G4ThreeVector   final_pos = post->GetPosition();
+  G4double        step_time = (pre->GetGlobalTime()  +
```
Comment (paolafer) [inline source/actions/SaveAllSteppingAction.cc:77]: Are you sure that you want the average time of the steps and not the time at the beginning or at the end? What do you use this for?

**Comment by paolafer** [comment]
`
// This class sets the store_steps flag of the persistency manager
// so all steps are saved.`

Could you improve a little the description of the class? It does other things beyond setting a flag.

File: source/actions/SaveAllSteppingAction.cc (hunk 1 above)
Comment (gonzaponte) [reply source/actions/SaveAllSteppingAction.cc:77]: I don't have a preference. I just want to have a handle on the time of the steps so I can potentially build a waveform out of them. Saving both pre and post seemed a bit too much, but I don't mind doing that and taking the average offline.

Comment (paolafer) [reply source/actions/SaveAllSteppingAction.cc:77]: It's just that in G4 the "borders" of a step are well defined in terms of position, volume... while what happens "inside" the step is sometimes more confuse. However, since yours is the only user case so far and you want the average, we can go ahead with that.

File: source/actions/SaveAllSteppingAction.cc [hunk 2]
```diff
@@ -1,8 +1,19 @@
 // ----------------------------------------------------------------------------
-// nexus | SaveAllSteppingAction.cc
+// nexus | SaveAllSteppingAction.h
```
Comment (paolafer) [inline source/actions/SaveAllSteppingAction.cc]: Why did you change the extension? It is `.cc`, right?

File: source/actions/SaveAllSteppingAction.cc [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+// pre- and post-step points, average time, process name and other identifiers)
+// of some steps of the simulation. By default all steps are stored. However,
+// a subset of them can be selected by cherry-picking the volumes and particles
+// involved in the step. This can be achieved with the commands
+// /Actions/SaveAllSteppingAction/select_particle
+// and
+// /Actions/SaveAllSteppingAction/select_volume
+// without the need for re-compilation.
+// It must be noted that the files produced with this action become large
+// very quicly. Therefore, strict filtering and small number of events are
```
Comment (paolafer) [inline source/actions/SaveAllSteppingAction.cc]: Typo: quicly -> quickly.

File: source/actions/SaveAllSteppingAction.cc (hunk 2 above)
Comment (gonzaponte) [reply source/actions/SaveAllSteppingAction.cc]: Yes, the wonders of copy-pasting...

**Review by paolafer — APPROVED** [review]
Useful addition in this stepping action, approved!
